src/args.rs

```diff
@@ -1,8 +1,115 @@
-#[derive(Debug)]
-pub struct Args {}
+use clap::Parser;
+
+#[derive(Parser, Debug, Default)]
+#[command(author, version, about, long_about = None, name = "jperf")]
+pub struct Args {
+	/// Filter results to just one app to monitor
+	#[arg(long, value_name = "application name")]
+	app: Option<String>,
+
+	/// Specify the Juju cloud where apps to monitor are deployed at
+	#[arg(long, value_name = "Juju cloud substrate")]
+	cloud: Option<String>,
+
+	/// Specify a single application unit to monitor
+	#[arg(long, value_name = "application unit name")]
+	unit: Option<String>,
+
+	/// Specify the Juju model name where apps to monitor are at
+	#[arg(long, value_name = "Juju model")]
+	model: Option<String>,		
+
+	/// Specify a single metric to display
+	#[arg(short, value_name = "metric name")]
+	metric: Option<String>,	
+	
+	/// Most recent time window in milliseconds to display
+	#[arg(long, default_value_t = 1000, value_name = "time interval")]
+	interval: u32,
+}
 
 impl Args {
-    pub fn parse() -> Self {
-        todo!()
-    }
+	#[allow(dead_code)]
+	fn app(&self) -> &Option<String>{
+		&self.app
+	}
+
+	#[allow(dead_code)]
+	fn unit(&self) -> &Option<String>{
+		&self.unit
+	}
+	
+	#[allow(dead_code)]
+	fn model(&self) -> &Option<String>{
+		&self.model
+	}
+
+	#[allow(dead_code)]
+	fn cloud(&self) -> &Option<String>{
+		&self.cloud
+	}
+
+	#[allow(dead_code)]
+	fn metric(&self) -> &Option<String>{
+		&self.metric
+	}
+
+	#[allow(dead_code)]
+	fn interval(&self) -> &u32 {
+		&self.interval
+	}
+
+	#[allow(dead_code)]
+	fn metric_as_string(&self) -> String {
+		match self.metric() {
+			None => "".to_string(),
+			Some(n) => n.to_string(),
+		}
+	}
+}
+
+#[cfg(test)]
+impl Args {
+	#[allow(dead_code)]
+	fn entity_as_string(&self, entity : &str) -> &Option<String> {		
+		match entity {
+			"cloud" => self.cloud(),
+			"model" => self.model(),
+			"app" => self.app(),
+			"unit" => self.unit(),
+			&_ => &None,
```

Comment:
The `&` in `&_` will be handled by the `_`, so I think it's slightly cleaner to make the reference implicit. I personally disagree with cargo's slightly more verbose suggestion which I realise this line originates from

Reply:
Changed to `_ => None,` .